#166272797 Implement babel and nodemon setup

adex001 opened the pull request:
#### What does this PR do?
- Implement babel and nodemon setup
#### Description of Task to be completed?
- install nodemon as a dev dependency
- install babel-node, babel-cli, babel-core, babel-preset-env, babel-register, morgan as a dependency
- create a .babelrc file and configure babel-preset-env
- create a start and start:dev script in the package.json
- ignore package-lock.json in .gitignore file
#### How should this be manually tested?
- Clone the repository, npm i to install all dependencies and npm start, then open Postman and navigate the endpoint.
#### Any background context you want to provide?
-  N/A
#### What are the relevant pivotal tracker stories?
[#166272797](https://www.pivotaltracker.com/story/show/166272797)
#### Screenshots (if appropriate)
- N/A
#### Questions:
- N/A

Commit:
chore-[166272797]: implement babel nodemon setup
- install nodemon as a dev dependency
- install babel-node, babel-cli, babel-core, babel-preset-env, babel-register, morgan as a dependency
- create a .babelrc file and configure babel-preset-env
- create a start and start:dev script in the package.json
- ignore package-lock.json in .gitignore file

fidelisojeah commented on .gitignore:
```diff
@@ -34,3 +34,4 @@ node_modules
 # Optional REPL history
 .node_repl_history
 
+package-lock.json
```
Remove this

adex001 replied:
Resolved this.

fidelisojeah commented on .babelrc:
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["@babel/preset-env"]
+}
```
`@babel/plugin-proposal-object-rest-spread`
`@babel/plugin-transform-arrow-functions`

the `@babel/preset-env` should target node

adex001 replied:
I have added the plugins and made sure @babel/preset-env targets node

fidelisojeah reviewed (changes requested):
Also consider updating some files: specifically `index.js` to use more es6+ like features like `import/export` and anonymous  functions

Commit:
chore-[166272797]: implement babel and nodemon setup
- refactor all require statements to make use of es6 syntax
- make @babel/preset-env target node as required in the PR
- add two plugins to the .babelrc file
- remove package-lock.json from the .gitignore file

fidelisojeah commented on config/index.js:
```diff
@@ -1,4 +1,4 @@
-module.exports = {
+export default {
```
perhaps:
```javascript
export const secret = process.env.NODE_ENV === 'production' ? process.env.SECRET : 'secret'
```
instead ?

fidelisojeah committed:
#166272797 - fix merge conflicts

Commit:
chore-[166272797]: implement babel and nodemon setup
- refactor ./config/index.js file for more readability

fidelisojeah reviewed (changes requested):
Fix linting issues

Commit:
Merge branch '166272797-ch-nodemon-babel-setup' of https://github.com/andela/kingsmen-ah-backend into 166272797-ch-nodemon-babel-setup
